### PR TITLE
ENH: Update ITKv4 registration example to use resample over warp

### PR DESCRIPTION
**Pull request by blowekamp**
The ResampleImageFilter is a more general filter than the
WarpImageFilter and should be preferred over the WarpImageFilter.

### CODE REVIEW

File: Examples/RegistrationITKv4/DeformableRegistration3.cxx [hunk 1]
```diff
@@ -290,53 +291,55 @@ int main( int argc, char *argv[] )
...
 
   // Software Guide : BeginLatex
   //
-  // Unlike the ResampleImageFilter, the WarpImageFilter
-  // warps or transforms the input image with respect to the deformation field
-  // represented by an image of vectors.  The resulting warped or resampled
-  // image is written to file as per previous examples.
+  // The displacement field is not an itk::Transform type. The
+  // ResampleImageFilter requires an itk::Transform as input, so a
+  // DisplacementFieldTransform need to be constructed. The
```
Comment (dzenanz) [inline Examples/RegistrationITKv4/DeformableRegistration3.cxx]: need -> needs

File: Examples/RegistrationITKv4/DeformableRegistration5.cxx [hunk 2]
```diff
@@ -289,56 +290,59 @@ int main( int argc, char *argv[] )
...
+
   // Software Guide : BeginLatex
   //
-  // Unlike the ResampleImageFilter, the WarpImageFilter
-  // warps or transforms the input image with respect to the deformation field
-  // represented by an image of vectors.  The resulting warped or resampled
-  // image is written to file as per previous examples.
+  // The displacement field is not an itk::Transform type. The
+  // ResampleImageFilter requires an itk::Transform as input, so a
+  // DisplacementFieldTransform need to be constructed. The
```
Comment (dzenanz) [inline Examples/RegistrationITKv4/DeformableRegistration5.cxx]: need -> needs

File: Examples/RegistrationITKv4/DeformableRegistration3.cxx [hunk 3]
```diff
@@ -290,53 +291,55 @@ int main( int argc, char *argv[] )
...
   // Software Guide : BeginLatex
   //
-  // The \doxygen{WarpImageFilter} can be used to warp the moving image with
-  // the output deformation field. Like the \doxygen{ResampleImageFilter},
-  // the WarpImageFilter requires the specification of the input image to be
-  // resampled, an input image interpolator, and the output image spacing and
-  // origin.
+  // The \doxygen{ResampleImageFilter} can be used to warp the moving image with
+  // the output deformation field. The \doxygen{ResampleImageFilter}
+  // requires the specification of the input image to be resampled, an
```
Comment (blowekamp) [inline Examples/RegistrationITKv4/DeformableRegistration3.cxx]: Hrmm... The old version was just as confusing. This needs to be clarified.

File: Examples/RegistrationITKv4/DeformableRegistration3.cxx (hunk 1 above)
Comment (blowekamp) [inline Examples/RegistrationITKv4/DeformableRegistration3.cxx]: needs to be...

**Pull request merged**